common/sagas/transactions.ts

```diff
@@ -34,6 +49,38 @@ export function* fetchTxData(action: FetchTransactionDataAction): SagaIterator {
   yield put(setTransactionData({ txhash, data, receipt, error }));
 }
 
+export function* saveBroadcastedTx(action: BroadcastTransactionQueuedAction) {
+  const txBuffer = action.payload.serializedTransaction;
+  const txIdx = action.payload.indexingHash;
+
+  const res: BroadcastTransactionSucceededAction | BroadcastTransactionFailedAction = yield take([
+    TxTypeKeys.BROADCAST_TRANSACTION_SUCCEEDED,
+    TxTypeKeys.BROADCAST_TRASACTION_FAILED
+  ]);
+
+  // If our TX succeeded, save it and update the store.
+  if (
+    res.type === TxTypeKeys.BROADCAST_TRANSACTION_SUCCEEDED &&
+    res.payload.indexingHash === txIdx
+  ) {
+    const tx = new EthTx(txBuffer);
+    saveRecentTransaction(res.payload.broadcastedHash, tx);
+    yield call(loadRecentTxs);
+  }
+}
+
+export function* loadRecentTxs() {
```

Comment:
confusing naming scheme here, maybe rename `loadRecentTransactions` to `loadRecentTransactionsFromLS`

Reply:
I use `tx` as shorthand in all of these saga function names, so I'll leave that for consistency, but I'll make it clear it's from LS.